internal/searcher/client.go

```diff
@@ -66,17 +68,68 @@ func (c *client) Resolve(pkg string) (*lock.Package, error) {
 	if len(result.Results) == 0 {
 		return nil, nix.ErrPackageNotFound
 	}
+
+	searchVersion := result.Results[0].Packages[0].Version
+	sysInfoMap := map[string]*lock.SystemInfo{}
+	if featureflag.RemoveNixpkgs.Enabled() {
+		// we use searchVersion instead of version so that "latest" is resolved
+		// to a concrete version before we get the package's system info
+		sysInfo, err := c.resolvePackageSystemInfoIfAny(name, searchVersion)
+		if err != nil {
+			return nil, err
+		}
+		if sysInfo != nil {
+			sysInfoMap[sysInfo.System] = sysInfo
+		}
+	}
 	return &lock.Package{
 		LastModified: result.Results[0].Packages[0].Date,
 		Resolved: fmt.Sprintf(
 			"github:NixOS/nixpkgs/%s#%s",
 			result.Results[0].Packages[0].NixpkgCommit,
 			result.Results[0].Packages[0].AttributePath,
 		),
-		Version: result.Results[0].Packages[0].Version,
+		Version: searchVersion,
+		Systems: sysInfoMap,
 	}, nil
 }
 
+// resolvePackageSystemInfoIfAny is temporary, until the search API returns
+// the "system info" like the store-hash. This uses the /pkg api that is
+// for nixhub.io as a temporary workaround.
+func (c *client) resolvePackageSystemInfoIfAny(pkgName, version string) (*lock.SystemInfo, error) {
+	packageResults, err := execPackageQuery(c.host, pkgName)
+	if err != nil {
+		return nil, err
+	}
+
+	var ok bool
+	result, ok := lo.Find(
+		packageResults, func(result *PackageResult) bool { return result.Version == version })
+	if !ok {
+		return nil, nil
+	}
+
+	userSystem, err := nix.System()
+	if err != nil {
+		return nil, err
+	}
+
+	var systemInfo *lock.SystemInfo
+	for sysName, sysInfo := range result.Systems {
+		if sysName == userSystem {
```

Comment:
Should we just add all of the systems we get back? There should be at most 4. It would make moving between local and cloud faster if your local system isn't x86 Linux.

Reply:
Actually, yes, we should. Good point.

It'll also simplify some of the "merge" logic I have elsewhere in this PR.